src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -8750,6 +8748,36 @@ public void M1()
         throw new System.NotImplementedException();
     }
 }
+");
+        }
+
+        [WorkItem(49019, "https://github.com/dotnet/roslyn/issues/49019")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestUnconstrainedGenericInstantiatedWithValueType()
+        {
+            await TestInRegularAndScriptAsync(@"
+interface IGoo<T>
+{
+    void Bar(T? x);
+}
+
+class C : [|IGoo<int>|]
+{
+}
+",
+@"
+interface IGoo<T>
+{
+    void Bar(T? x);
+}
+
+class C : IGoo<int>
+{
+    public void Bar(int x)
+    {
+        throw new System.NotImplementedException();
+    }
+}
 ");
         }
```

Comment:
For completeness, you may want to add another test with `where T : struct` constraint:

```suggestion
        }

        [WorkItem(49019, "https://github.com/dotnet/roslyn/issues/49019")]
        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
        public async Task TestConstrainedGenericInstantiatedWithValueType()
        {
            await TestInRegularAndScriptAsync(@"
interface IGoo<T> where T : struct
{
    void Bar(T? x);
}
class C : [|IGoo<int>|]
{
}
",
@"
interface IGoo<T> where T : struct
{
    void Bar(T? x);
}
class C : IGoo<int>
{
    public void Bar(int? x)
    {
        throw new System.NotImplementedException();
    }
}
");
        }
```

Reply:
done.